internal/database/database.go

```diff
@@ -1172,9 +1185,13 @@ func (d *Database) UpsertSource(ctx context.Context, principal string, source *c
 				return err
 			}
 
-			if err := d.upsert(ctx, tx, "sources_datasources", []string{"source_name", "name", "type", "path", "config", "transform_query"},
+			var secret sql.NullString
+			if datasource.Credentials != nil {
+				secret.String, secret.Valid = datasource.Credentials.Name, true
```

Comment:
Funny there is no helper factory function for creating a `sql.NullString` that correctly assigns the `Valid` field. Suppose this type is mostly for reading data(?).